src/components/date_picker/super_date_picker/quick_select_popover/quick_select_utils.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * This function returns time value, time unit and time tense for a given time string.
+ * For example: for `now-40m` it will parse output as time value to `40`
+ * time unit to `m` and time unit to `last`.
+ * If given a datetime string it will return a default value.
+ * If the given string is in the format such as `now/d` it will parse the string to moment object
+ * and find the time value, time unit and time tense using moment
+ * This function accepts two strings start and end time. I the start value is now then it uses
+ * the end value to parse.
+ *
+ * @param {string} start start time
+ * @param {string} start end time
+ * @returns {object} time value, time unit and time tense
+ */
+
+import moment from 'moment';
+import dateMath from '@elastic/datemath';
+import { isString } from '../../../../services/predicate';
+import { relativeUnitsFromLargestToSmallest } from '../relative_options';
+import { DATE_MODES } from '../date_modes';
+
+const LAST = 'last';
+const NEXT = 'next';
+
+const isNow = value => value === DATE_MODES.NOW;
+
+export const parseTimeParts = (start, end) => {
+  const results = {
+    timeValueDefault: 15,
+    timeUnitsDefault: 'm',
+    timeTenseDefault: LAST,
+  };
+
+  const value = isNow(start) ? end : start;
+
+  const matches =
+    isString(value) &&
+    value.match(/now(([\-\+])([0-9]+)([smhdwMy])(\/[smhdwMy])?)?/);
+
+  if (!matches) {
+    return results;
+  }
+
+  const operator = matches && matches[2];
+  const timeValue = matches && matches[3];
+  const timeUnitsDefault = matches && matches[4];
+
+  if (timeValue && timeUnitsDefault && operator) {
+    const timeValueDefault = parseInt(timeValue);
```

Comment:
`parseInt` should always be passed the radix operator as the second parameter. For this component, it *should* be impossible to pass strings that indicate non-base-10 numbers, but better to be defensive.

```suggestion
    const timeValueDefault = parseInt(timeValue, 10);
```